src/main/java/org/folio/circulation/services/agedtolost/AssignLostFeesWhenAgedToLostService.java

```diff
@@ -0,0 +1,286 @@
+  private final ItemRepository itemRepository;
+
+  public AssignLostFeesWhenAgedToLostService(Clients clients) {
+    this.lostItemPolicyRepository = new LostItemPolicyRepository(clients);
+    this.feeFineOwnerRepository = new FeeFineOwnerRepository(clients);
+    this.feeFineRepository = new FeeFineRepository(clients);
+    this.feeFineFacade = new FeeFineFacade(clients);
+    this.loanRepository = new LoanRepository(clients);
+    this.itemRepository = new ItemRepository(clients, true, false, false);
+  }
+
+  public CompletableFuture<Result<Void>> processFeeAssigning() {
+    return fetchLoansAndItems()
+      .thenCompose(r -> r.after(allLoans -> {
+        if (allLoans.isEmpty()) {
+          log.info("No aged to lost loans to assign lost fees");
+          return completedFuture(succeeded(null));
+        }
+
+        log.debug("Loans to assign fees [{}]", allLoans.getRecords().size());
+
+        return succeeded(LoanToAssignFees.usingLoans(allLoans))
+          .after(this::fetchFeeFineOwners)
+          .thenComposeAsync(this::fetchFeeFineTypes)
+          .thenCompose(this::assignLostFeesForLoans);
+      }));
+  }
+
+  private CompletableFuture<Result<Void>> assignLostFeesForLoans(
+    Result<List<LoanToAssignFees>> loansToAssignFeesResult) {
+
+    return loansToAssignFeesResult
+      .after(loansToAssignFees -> allOf(loansToAssignFees, this::assignLostFeesForLoan))
+      .thenApply(r -> r.map(notUsed -> null));
+  }
+
+  private CompletableFuture<Result<Loan>> assignLostFeesForLoan(LoanToAssignFees loanToAssignFees) {
+    return createAccountsForLoan(loanToAssignFees)
+      .after(feeFineFacade::createAccounts)
+      .thenCompose(r -> r.after(notUsed -> updateLoanBillingInfo(loanToAssignFees)));
+  }
+
+  private CompletableFuture<Result<Loan>> updateLoanBillingInfo(LoanToAssignFees loanToAssignFees) {
+    final Loan updatedLoan = loanToAssignFees.getLoan().setLostItemHasBeenBilled();
+
+    return loanRepository.updateLoan(updatedLoan);
+  }
+
+  private Result<List<CreateAccountCommand>> createAccountsForLoan(LoanToAssignFees loanToAssignFees) {
+    return validateCanCreateAccountForLoan(loanToAssignFees)
+      .map(notUsed -> {
+        final LostItemPolicy lostItemPolicy = loanToAssignFees.getLostItemPolicy();
+
+        return Stream.of(pair(lostItemPolicy.getSetCostFee(), loanToAssignFees.getLostItemFeeType()),
+            pair(lostItemPolicy.getProcessingFee(), loanToAssignFees.getLostItemProcessingFeeType()))
+          .filter(pair -> pair.getKey().isChargeable())
+          .map(pair -> CreateAccountCommand.builder()
+            .withAmount(pair.getKey().getAmount())
+            .withCreatedByAutomatedProcess(true)
+            .withFeeFine(pair.getValue())
+            .withFeeFineOwner(loanToAssignFees.getOwner())
+            .withLoan(loanToAssignFees.getLoan())
+            .withItem(loanToAssignFees.getLoan().getItem())
+            .build())
+          .collect(Collectors.toList());
+      });
```

Comment:
Could some methods be extracted here to convey the intent of the steps better. I'm not sure I follow what this does.

What does the stream and filter do? Maybe a method describing what that does and a method for creating the account?